js/src/api/contract/contract.js

```diff
@@ -139,6 +139,7 @@ export default class Contract {
             }
 
             setState({ state: 'hasReceipt', receipt });
+            this._receipt = receipt;
```

Comment:
Anything with `_` is private and should not be accessed outside of the module at all, i.e. the implementation can change completely and it should only affect the module, in this case the API. 

So setting it here and accessing it in a modal is  a no-go area.